tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/OsgiBundleProject.java

```diff
@@ -593,4 +603,26 @@ private void applyBestOfCurrentOrConfiguredProfile(String configuredProfileName,
             sink.setProfileConfiguration(configuredProfileName, reason);
         }
     }
+
+    @Override
+    public List<ClasspathEntry> getTestClasspath(ReactorProject reactorProject) {
+        @SuppressWarnings("unchecked")
+        List<ClasspathEntry> classpath = (List<ClasspathEntry>) reactorProject
+                .getContextValue(TychoConstants.CTX_ECLIPSE_PLUGIN_TEST_CLASSPATH);
+        if (classpath == null) {
+            throw new IllegalStateException();
+        }
+        return classpath;
+    }
+
+    public DependencyArtifacts getTestDependencyArtifacts(ReactorProject project) {
+        return TychoProjectUtils.getTestDependencyArtifacts(project);
+    }
+
+    @Override
+    public List<Dependency> getExtraTestRequirements() {
+        // TODO parse .classpath to resolve "test" scope deps in it and feed those extra dependencies
+        return Collections.emptyList();
```

Comment:
Actually "Dependency" might be a package with version range or a capability, how would this be expressed?

Reply:
There is currently nothing that allows to express that in Tycho "extraRequirements" as far as I know. The issue seems relatively independent technically (ie we could already start supporting
```xml
  <extraRequirement>
    <type>osgi.wiring.package</type> <!-- or java.package to use p2 wording -->
    <id>com.acme.pack</id>
    <version>0.0.0</version>
  <extraRequirement>
```
and use the same thing for extra test requirements generated from containers or whatever.

I think this story of expliciting requirement on packages deserves its own issue and PR, and then we can easily join both.

Reply:
If we can extend this to generic namespace that would be useful. For example to pull in DS one should describe the requirement in the following forms:

```
Require-Capability: osgi.extender;
  filter:="(&(osgi.extender=osgi.component)(version>=1.4)(!(version>=2.0)))"
```
http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-capabilities

In P2 this is expressed by the org.eclipse.equinox.internal.p2.metadata.IRequiredCapability requirement and you can require other "P2" properties to (e.g. ws, os, ...)